.github/workflows/main.yaml

```diff
@@ -261,16 +261,24 @@ jobs:
           target: ${{ matrix.job.target }}
 
   build_envoy:
-    runs-on: neco-containers-ubuntu-20.04
+    runs-on: neco-containers-ubuntu-22.04
     steps:
       - uses: actions/checkout@v4
+      - name: Read TAG
+        run: |
+          TAG=$(cat ./envoy/TAG)
+          echo "TAG=${TAG}" >> $GITHUB_ENV
+          VERSION=$(echo $TAG | cut -d \. -f 1,2,3)
+          echo "VERSION=${VERSION}" >> $GITHUB_ENV
       - name: Build container image
         uses: ./.github/actions/build_envoy
         with:
-          version: 1.28.1
+          version: ${{ env.VERSION }}
```

Comment:
The version parameter can be completely deleted if we parse the TAG file in the `build_envoy` job.